src/oidc.rs

```diff
@@ -47,8 +43,11 @@ pub fn request(app: &AppConfig, params: &QueryMap) -> Url {
     // Retrieve the provider's Discovery document and extract the
     // `authorization_endpoint` from it.
     let provider = &app.providers[&email_addr.domain];
-    let val = fetch_json_cached(app, &provider.discovery);
-    let config = val.as_object().unwrap();
+    let config_val: Value = {
```

Comment:
These separate scopes are a good idea, but please put them in a separate commit from the one that introduces the caching methods, so that the changes for each commit are clearly described by the commit message.